packages/devtools_app/lib/src/shared/memory/heap_graph_loader.dart

```diff
@@ -44,3 +46,17 @@ class HeapGraphLoaderFile implements HeapGraphLoader {
     );
   }
 }
+
+/// Loads a heap snapshot from `List<ByteData>` created with HeapSnapshotGraph.toChunks.
+class HeapGraphLoaderFromChunks implements HeapGraphLoader {
+  HeapGraphLoaderFromChunks({required this.chunks, required this.created});
+
+  List<ByteData> chunks;
+
+  DateTime created;
+
+  @override
+  Future<(HeapSnapshotGraph, DateTime)> load() async {
+    return (HeapSnapshotGraph.fromChunks(chunks), created);
+  }
```

Comment:
this doesn't need to be async. If the super class returns FutureOr, then we can remove the unnecessary async here